openassessment/xblock/static/js/src/lms/containers/WaitingStepDetailsContainer.jsx

```diff
@@ -20,15 +20,45 @@ const WaitingStepDetailsContainer = ({ waitingStepDataUrl, onMount }) => {
     setError(false);
 
     // Make request using API
-    const { success, waitingStepData } = await fetchWaitingStepDetails(waitingStepDataUrl);
+    try {
+      // Make request using API
+      const { success, waitingStepData } = await fetchWaitingStepDetails(waitingStepDataUrl);
 
-    // Check response and save contents on state
-    if (success) {
-      setWaitingStepDetails(waitingStepData);
-    } else {
+      // Check response and save contents on state
+      if (success) {
+        setWaitingStepDetails(waitingStepData);
+      } else {
+        setError(true);
+      }
+    } catch (err) {
       setError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const getUserNamSelected = (username) => {
+    const button = document.querySelector('.button-staff-tools');
```

Comment:
is querySelectos the best way of getting the user's info?

Reply:
What if the classes or attributes are changed?

Reply:
We would have to re do the component that carry the data of the student, remember that this has django template mixed and make other request

Reply:
I don't think so that can be changed, those classes are  in css code and js code

Reply:
Is this pattern of using querySelector used elsewhere in this repo?

Reply:
The React part is something new, other parts are using Jquery that is not an option